compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -1664,7 +1883,10 @@ abstract class GenJSCode extends plugins.PluginComponent
       if (sym.owner == StringClass && !isStringMethodFromObject) {
         genStringCall(tree)
       } else if (isRawJSType(receiver.tpe) && sym.owner != ObjectClass) {
-        genPrimitiveJSCall(tree, isStat)
+        if (!isScalaJSDefinedJSClass(sym.owner) || jsInterop.isExposed(sym))
+          genPrimitiveJSCall(tree, isStat)
```

Comment:
Fixed by having `jsInterop.{isExposed,registerJSExposedSymbol}` manipulate annotations under the hood.